python/ray/serve/backend_state.py

```diff
@@ -275,7 +275,8 @@ class BackendReplica(VersionedReplica):
 
     def __init__(self, controller_name: str, detached: bool,
                  replica_tag: ReplicaTag, backend_tag: BackendTag,
-                 version: BackendVersion):
+                 version: BackendVersion,
+                 state: Optional[ReplicaState] = ReplicaState.SHOULD_START):
```

Comment:
I wonder if we should just pull out the state from the `BackendReplica` before this change entirely. Thoughts?

Reply:
let me look into this. the "state" field in this class didn't really make sense to me for the most part anyways ... and it's required to change for the context of this PR since it fails assertions.